neopixel.py

```diff
@@ -85,23 +95,33 @@ def __init__(self, num_leds, state_machine, pin, mode="RGB", delay=0.0001):
         :param delay: [default: 0.0001] delay used for latching of leds when sending data
         """
         self.pixels = array.array("I", [0] * num_leds)
+        # self.pixels_out = array.array("I", [0] * num_leds)
         self.mode = mode
         self.W_in_mode = 'W' in mode
+        bpp = 3
         if self.W_in_mode:
             # RGBW uses different PIO state machine configuration
-            self.sm = rp2.StateMachine(state_machine, sk6812, freq=8000000, sideset_base=Pin(pin))
+            self.sm = rp2.StateMachine(state_machine, sk6812, freq=12_800_000, sideset_base=Pin(pin))
             # tuple of values required to shift bit into position (check class desc.)
             self.shift = ((mode.index('R') ^ 3) * 8, (mode.index('G') ^ 3) * 8,
                           (mode.index('B') ^ 3) * 8, (mode.index('W') ^ 3) * 8)
+            bpp = 4
         else:
-            self.sm = rp2.StateMachine(state_machine, ws2812, freq=8000000, sideset_base=Pin(pin))
+            self.sm = rp2.StateMachine(state_machine, ws2812, freq=12_800_000, sideset_base=Pin(pin))
             self.shift = (((mode.index('R') ^ 3) - 1) * 8, ((mode.index('G') ^ 3) - 1) * 8,
                           ((mode.index('B') ^ 3) - 1) * 8, 0)
         self.sm.active(1)
         self.num_leds = num_leds
         self.delay = delay
         self.brightnessvalue = 255
 
+        byte_count = bpp * num_leds
+        bit_count = byte_count * 8
+        padding_count = -byte_count % 4
+        self.header = bytearray(struct.pack("L", bit_count - 1))
+        self.trailer = bytearray(b"\0" * padding_count + struct.pack("L", 3840))
```

Comment:
A comment here would be nice so we know why exactly this is needed in further development.